pkg/networkservice/common/connect/server_test.go

```diff
@@ -262,16 +263,22 @@ func TestConnectServer_RequestParallel(t *testing.T) {
 	wg.Wait()
 	wg.Add(parallelCount)
 
-	assert.Equal(t, int32(parallelCount), serverClient.count)
-	assert.Equal(t, int32(parallelCount), serverA.count)
-	assert.Equal(t, int32(parallelCount), serverNext.count)
+	assert.Equal(t, int32(parallelCount), atomic.LoadInt32(&serverClient.Requests))
+	assert.Equal(t, int32(0), atomic.LoadInt32(&serverClient.Closes))
+	assert.Equal(t, int32(parallelCount), atomic.LoadInt32(&serverA.Requests))
+	assert.Equal(t, int32(0), atomic.LoadInt32(&serverA.Closes))
+	assert.Equal(t, int32(parallelCount), atomic.LoadInt32(&serverNext.Requests))
+	assert.Equal(t, int32(0), atomic.LoadInt32(&serverNext.Closes))
```

Comment:
Why do we use `assert` here (and below), instead of `require`?

Reply:
Here we need to use `assert` before `273:    barrier.Done()`, because `require` will lead us to goroutines locked on `barrier` and so lots of errors.
But here is my fault, after this we can continue using `require` and so there is no need for any `assert`.

Fixed L276-L281 to use only `require`.

Reply:
I think it would help to add a comment about this before these asserts, to prevent people from trying to refactor it in the future.

Reply:
Added comments near `barrier` initialization and near `barrier.Done()`.